signer/src/logging.rs

```diff
@@ -43,3 +48,53 @@ fn setup_logging_pretty(directives: &str) {
         .with(main_layer)
         .init()
 }
+
+/// Logs to standard logging stream information about Bitcoin and Stacks
+/// node versions, chaintips, dkg rounds, etc.
+async fn log_blockchain_nodes_info<C: Context>(ctx: &C) {
+    let info = build_info(ctx).await;
+    tracing::debug!(?info, "logging blockchain info",);
+}
+
+/// Simple struct for time to time writing logs
+/// about Stacks and Bitcoin nodes state.
+pub struct BlockchainInfoLogger<Context> {
+    /// Signer context.
+    context: Context,
+    /// Logging period.
+    timeout: Duration,
+}
+
+impl<C> BlockchainInfoLogger<C>
+where
+    C: Context,
+{
+    /// Creates new BlockchainInfoLogger with given context and timeout.
+    pub fn new(context: C, timeout: Duration) -> Self {
+        Self { context, timeout }
+    }
+    /// Runs BlockchainInfoLogger which will log info about blockchain nodes
+    /// each timeout.
+    pub async fn run(self) {
+        let term = self.context.get_termination_handle();
+
+        // Without this counter and each_nth_iteration_print_log,
+        // if self.timeout is big, tokio::time::sleep will sleep for
+        // full timeout, and it will not check if termination was activated
+        // for a long time.
+        let each_nth_iteration_print_log = self.timeout.as_secs();
```

Comment:
We can use a `tokio::select!`, we do something similar for the p2p event loop (https://github.com/stacks-sbtc/sbtc/blob/9507bd4a41bee4cf038fa30f02628df1aeb14bb9/signer/src/network/libp2p/event_loop.rs#L255-L256)

Reply:
[59a3ffe](https://github.com/stacks-sbtc/sbtc/pull/1787/commits/59a3ffe72b3883c1a7477c424623a2b27dc11a38)